### PR TITLE
Support for python packages

### PR DESCRIPTION
**Pull request by oxisto**
This PR adds (basic) support for python *packages* in addition to python *modules*.

### CODE REVIEW

**Comment by oxisto** [comment]
This is unfortunately blocked until we have some sort of package dependency graph.

**Comment by oxisto** [comment]
Depends on #1782 
Depends on #1783

**Comment by oxisto** [comment]
This should work even better once the blocked PRs are merged in, but this can already be reviewed independently I think.

**Review by KuechA — APPROVED** [review]

**Pull request merged**
